pkg/deploy/lattice/service_network_manager.go

```diff
@@ -54,61 +58,61 @@ type defaultServiceNetworkManager struct {
 // return errors.New(LATTICE_RETRY) when:
 //
 //	CreateServiceNetworkVpcAssociationInput returns ServiceNetworkVpcAssociationStatusFailed/ServiceNetworkVpcAssociationStatusCreateInProgress/MeshVpcAssociationStatusDeleteInProgress
-func (m *defaultServiceNetworkManager) Create(
-	ctx context.Context,
-	serviceNetwork *latticemodel.ServiceNetwork,
-) (latticemodel.ServiceNetworkStatus, error) {
+
+func (m *defaultServiceNetworkManager) CreateOrUpdate(ctx context.Context, desiredSn *latticemodel.ServiceNetwork) (latticemodel.ServiceNetworkStatus, error) {
 	// check if exists
-	serviceNetworkSummary, err := m.cloud.Lattice().FindServiceNetwork(ctx, serviceNetwork.Spec.Name, "")
+	foundSnSummary, err := m.cloud.Lattice().FindServiceNetwork(ctx, desiredSn.Spec.Name, "")
 	if err != nil && !services.IsNotFoundError(err) {
 		return latticemodel.ServiceNetworkStatus{ServiceNetworkARN: "", ServiceNetworkID: ""}, err
 	}
 
 	// pre declaration
 	var serviceNetworkId string
 	var serviceNetworkArn string
-	var isServiceNetworkAssociatedWithVpc bool
-	var serviceNetworkAssociatedWithCurrentVpcId *string
+	var isSnAlreadyAssociatedWithCurrentVpc bool
+	var snvaAssociatedWithCurrentVPC *vpclattice.ServiceNetworkVpcAssociationSummary
 	vpcLatticeSess := m.cloud.Lattice()
-
-	if serviceNetworkSummary == nil {
+	if foundSnSummary == nil {
 		m.log.Debugf("Creating ServiceNetwork %s and tagging it with vpcId %s",
-			serviceNetwork.Spec.Name, config.VpcID)
-
+			desiredSn.Spec.Name, config.VpcID)
 		// Add tag to show this is the VPC create this service network
 		// This means, the service network can only be deleted by the controller running in this VPC
 		serviceNetworkInput := vpclattice.CreateServiceNetworkInput{
-			Name: &serviceNetwork.Spec.Name,
+			Name: &desiredSn.Spec.Name,
 			Tags: make(map[string]*string),
 		}
 		serviceNetworkInput.Tags[latticemodel.K8SServiceNetworkOwnedByVPC] = &config.VpcID
 
-		m.log.Debugf("Creating serviceNetwork %+v", serviceNetworkInput)
+		m.log.Debugf("Creating desiredSn %+v", serviceNetworkInput)
 		resp, err := vpcLatticeSess.CreateServiceNetworkWithContext(ctx, &serviceNetworkInput)
 		if err != nil {
 			return latticemodel.ServiceNetworkStatus{ServiceNetworkARN: "", ServiceNetworkID: ""}, err
 		}
 
 		serviceNetworkId = aws.StringValue(resp.Id)
 		serviceNetworkArn = aws.StringValue(resp.Arn)
-		isServiceNetworkAssociatedWithVpc = false
 	} else {
-		glog.V(6).Infof("serviceNetwork %s exists, checking its VPC association", serviceNetwork.Spec.Name)
-		serviceNetworkId = aws.StringValue(serviceNetworkSummary.SvcNetwork.Id)
-		serviceNetworkArn = aws.StringValue(serviceNetworkSummary.SvcNetwork.Arn)
-		isServiceNetworkAssociatedWithVpc, serviceNetworkAssociatedWithCurrentVpcId, _, err = m.isServiceNetworkAssociatedWithVPC(ctx, serviceNetworkId)
+		m.log.Debugf("desiredSn %s exists, checking its VPC association", desiredSn.Spec.Name)
```

Comment:
Seems to have overwritten "service network" with "desiredSn"

Reply:
It did a variable rename from `serviceNetwork` to `desiredSn `? do you also think it's not a good renaming?